### PR TITLE
[DBParameterGroup] Include parameters in the `ReadHandler` response

### PR DESCRIPTION
**Pull request by osdrv**
This commit adds `DBParameterGroup.parameters` in the `ReadHandler` response. The motivation for this change is to enable drift detector on independent group parameters. At the moment this collection is defined as `WriteOnly`, instructing the drift detector to ignore it completely.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/LoggingProxyClient.java [hunk 1]
```diff
@@ -76,5 +79,19 @@ private <RequestT extends AwsRequest, ResultT> ResultT logAndDelegate(
...
+        ResultT result = null;
+        try {
+            requestLogger.log(request);
+            result = injectCredentials.apply(request, requestFunction);
+        } catch (Exception e) {
+            requestLogger.logAndThrow(e);
+        }
+        requestLogger.log("[Result log omitted]");
+        return result;
+    }
```
Comment (osdrv) [inline aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/LoggingProxyClient.java:96]: Logging the response here and in `AmazonWebServicesClientProxy.logRequestMetadataV2` (See https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/issues/414 for more details) caused our paginators to execute 3x more requests than needed.

**Review by moataz-mhmd — CHANGES_REQUESTED** [review]
`computeModifiedDBParameters` will miss out of band changes for parameters that stack contains. 
Example: a parameter `A` with value `Not_default` and out of band change put parameter `A` to value `default`
Suggesting to always add the parameters that contained in stack whatever values it holds beside the computed modified parameters.

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java [hunk 2]
```diff
@@ -383,97 +392,175 @@ private Map<String, Parameter> getParametersToReset(
...
+
+    @VisibleForTesting
+    static Map<String, Parameter> computeModifiedDBParameters(
+            @NonNull final Map<String, Parameter> engineDefaultParameters,
+            @NonNull final Map<String, Parameter> currentDBParameters
+    ) {
+        final Map<String, Parameter> modifiedParameters = new HashMap<>();
+        for (final String paramName : currentDBParameters.keySet()) {
+            final Parameter currentParam = currentDBParameters.get(paramName);
+            final Parameter defaultParam = engineDefaultParameters.get(paramName);
```
Comment (moataz-mhmd) [inline aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java:558]: Can we calculate and log delta between `currentDBParameters` and `engineDefaultParameters`. Is happy scenario. It should be empty, However we saw some added attributes without being in Deafult parameters due to nature of engine. We need to have those explicitly in our logs if there any.

**Review by moataz-mhmd — APPROVED** [review]

File: aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/LoggingProxyClient.java (hunk 1 above)
Comment (khebul) [reply aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/LoggingProxyClient.java:96]: Does only logging request records RequestId?

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/ReadHandler.java [hunk 3]
```diff
@@ -44,7 +47,11 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                     try {
                         final DBParameterGroup dBParameterGroup = describeResponse.dbParameterGroups().stream().findFirst().get();
                         context.setDbParameterGroupArn(dBParameterGroup.dbParameterGroupArn());
-                        return ProgressEvent.progress(Translator.translateFromDBParameterGroup(dBParameterGroup), context);
+                        final ResourceModel currentModel = Translator.translateFromDBParameterGroup(dBParameterGroup);
+                        if (model.getParameters() != null) {
```
Comment (khebul) [inline aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/ReadHandler.java:51]: Combining this line with https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/406/files#diff-271db357c6f3b9c5912af25ea427a66424f11ba56050a7f16099dab07166cdecR64 
Does describe response returns parameters? In which case the parameters would be missing?

**Review by khebul — APPROVED** [review]

**Review by moataz-mhmd — APPROVED** [review]

**Pull request merged**